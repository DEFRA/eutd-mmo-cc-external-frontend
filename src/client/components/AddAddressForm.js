import React, { Component } from 'react';
import {
  InputField,
  GridRow,
  GridCol,
  Label
} from 'govuk-react';

import SecondaryButton from './elements/SecondaryButton';
import AccessibleAutoComplete from '../components/AccessibleAutocomplete';
import ErrorText from '@govuk-react/error-text';
import {withTranslation} from 'react-i18next';

export default withTranslation() (class AddressForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      subBuildingName: '',
      buildingNumber: '',
      buildingName: '',
      streetName: '',
      townCity: '',
      county: '',
      postcode: '',
      country: ''
    };
  }

  onChange = (e) => {
    this.setState({
      [e.target.name]: (e.target.value),
    });
  };

  countrySearch = (countries) => (searchQuery) => {
    return countries.filter(country => country.toLowerCase().includes(searchQuery.toLowerCase()));
  };

  onConfirm = (country) => {
    if (!country) {
      let el = document.getElementById('country');
      if (el) {
        this.setState({
          country: el.value
        });
      }
      return;
    }
    this.setState({
      country
    });
  };

  componentDidMount() {
    const { preSelected } = this.props;

    this.setState({
      buildingNumber: preSelected ? preSelected.building_number : '',
      subBuildingName: preSelected ? preSelected.sub_building_name : '',
      buildingName: preSelected ? preSelected.building_name : '',
      streetName: preSelected ? preSelected.street_name : '',
      townCity: preSelected ? preSelected.city : '',
      county: preSelected ? preSelected.county : '',
      postcode: preSelected ? preSelected.postCode : '',
      country: preSelected ? preSelected.country : ''
    });
  }

  render() {
    const {
      subBuildingName,
      buildingNumber,
      buildingName,
      streetName,
      townCity,
      county,
      postcode,
      country,
    } = this.state;

    const { onSubmit, errors, countries, t } = this.props;

    return (
      <form onSubmit={(e) => {
        e.preventDefault();
        onSubmit(this.state);
      }}>
        <GridRow>
          <GridCol>
            <InputField
              meta={{
                error: t(errors.subBuildingNameError),
                touched: true
              }}
              htmlFor={'subBuildingName'}
              input={{
                id: 'subBuildingName',
                className: 'mediumInput',
                name: 'subBuildingName',
                value: subBuildingName,
                onChange: e => this.onChange(e),
                onBlur: e => this.onChange(e)
              }}
              hint={t('commonWhatExportersAddressSubBuildingNameHint')}
            >
              {t('commonWhatExportersAddressSubBuildingName')}
            </InputField>
          </GridCol>
        </GridRow>
        <GridRow>
          <GridCol>
            <InputField
              meta={{
                error: t(errors.buildingNumberError),
                touched: true
              }}
              htmlFor={'buildingNumber'}
              input={{
                id: 'buildingNumber',
                className: 'smallInput',
                name: 'buildingNumber',
                value: buildingNumber,
                onChange: e => this.onChange(e),
                onBlur: e => this.onChange(e)
              }}
            >
            {t('commonWhatExportersAddressBuildingNumber')}
            </InputField>
          </GridCol>
        </GridRow>
        <GridRow>
          <GridCol>
            <InputField
              meta={{
                error: t(errors.buildingNameError),
                touched: true
              }}
              htmlFor={'buildingName'}
              input={{
                id: 'buildingName',
                className: 'mediumInput',
                name: 'buildingName',
                value: buildingName,
                onChange: e => this.onChange(e),
                onBlur: e => this.onChange(e)
              }}
            >
              {t('commonWhatExportersAddressBuildingName')}
            </InputField>
          </GridCol>
        </GridRow>
        <GridRow>
          <GridCol >
            <InputField
              meta={{
                error: t(errors.streetNameError),
                touched: true
              }}
              htmlFor={'streetName'}
              input={{
                id: 'streetName',
                className: 'mediumInput',
                name: 'streetName',
                value: streetName,
                onChange: e => this.onChange(e),
                onBlur: e => this.onChange(e)
              }}
            >
               {t('commonWhatExportersAddressStreetName')}
            </InputField>
          </GridCol>
        </GridRow>
        <GridRow>
          <GridCol >
            <InputField
              meta={{
                error: t(errors.townCityError),
                touched: true
              }}
              htmlFor={'townCity'}
              input={{
                id: 'townCity',
                className: 'mediumInput',
                name: 'townCity',
                value: townCity,
                onChange: e => this.onChange(e),
                onBlur: e => this.onChange(e)
              }}
            >
              {t('commonWhatExportersAddressTownOrCity')}
            </InputField>
          </GridCol>
        </GridRow>
        <GridRow>
          <GridCol>
            <InputField
              meta={{
                error: t(errors.countyError),
                touched: true
              }}
              htmlFor={'county'}
              input={{
                id: 'county',
                className: 'mediumInput',
                name: 'county',
                value: county,
                onChange: e => this.onChange(e),
                onBlur: e => this.onChange(e)
              }}
            >
              {t('commonWhatExportersAddressCountyStateProvince')}
            </InputField>
          </GridCol>
        </GridRow>
        <GridRow>
          <GridCol>
            <InputField
              meta={{
                error: t(errors.postcodeError),
                touched: true
              }}
              htmlFor={'postcode'}
              input={{
                id: 'postcode',
                className: 'smallInput',
                name: 'postcode',
                value: postcode,
                onChange: e => this.onChange(e),
                onBlur: e => this.onChange(e)
              }}
            >
              {t('commonWhatExportersAddressPostcode')}
            </InputField>
          </GridCol>
        </GridRow>
        <GridRow>
          <GridCol >
            <Label htmlFor={'Country'} error={errors.countryError}>
              {t('commonWhatExportersAddressCountry')}
              <div style={{ maxWidth: '23em' }}>
                {errors && <ErrorText>{t(errors.countryError)}</ErrorText>}
                <AccessibleAutoComplete
                  error={errors.countryError}
                  id={'country'}
                  name='country'
                  defaultSelectMessage={''}
                  defaultValue={country || ''}
                  nojsValues={countries}
                  displayMenu="overlay"
                  search={this.countrySearch(countries)}
                  onConfirm={this.onConfirm}
                />

              </div>
            </Label>

          </GridCol>
        </GridRow>

        <GridRow>
          <SecondaryButton
            type="button"
            id="cancel"
            name="cancel"
            value="cancel"
            onClick={this.props.handleCancelButton}
          >
            {t('commonSecondaryButtonCancelButton')}
          </SecondaryButton>
          <GridCol>
            <button
              id="submit"
              className="button"
            >
             {t('commonContinueButtonContinueButtonText')}
            </button>
          </GridCol>
        </GridRow>
      </form>
    );
  }
});

