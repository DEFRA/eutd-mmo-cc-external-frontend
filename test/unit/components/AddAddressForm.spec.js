import React from 'react';
import { mount } from 'enzyme';
import AddAddressForm from '../../../src/client/components/AddAddressForm';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import thunk from 'redux-thunk';
import { render } from '@testing-library/react';

describe('AddAddressForm', () => {

  const mockStore = configureStore([thunk.withExtraArgument({
    orchestrationApi: {
      get: () => {
        return new Promise((res) => {
          res({});
        });
      },
      post: () => {
        return new Promise((res) => {
          res({});
        });
      }
    }
  })]);

  const store = mockStore({
    preSelected: {
      building_name: 'LANCASTER HOUSE',
      city: 'NEWCASTLE UPON TYNE',
      country: 'ENGLAND',
      county: 'TYNESIDE',
      postCode: 'NE4 7YH',
      street_name: 'HAMPSHIRE COURT',
      sub_building_name: 'sub building name',
      building_number: 'building number'
    }
  });

  const props = {
    countries: ['Brazil', 'UK'],
    errors: {},
    preSelected: {
      building_name: 'LANCASTER HOUSE',
      city: 'NEWCASTLE UPON TYNE',
      country: 'ENGLAND',
      county: 'TYNESIDE',
      postCode: 'NE4 7YH',
      street_name: 'HAMPSHIRE COURT',
      sub_building_name: 'sub building name',
      building_number: 'building number'
    },
    handleCancelButton: jest.fn()
  };
  const wrapper = mount(
    <Provider store={store}>
        <MemoryRouter>
          <AddAddressForm {...props}/>
        </MemoryRouter>
      </Provider>

    );

  it('Should render the AddAddressForm component', () => {
    expect(wrapper).toBeDefined();

  });

  it('should have an id on all inputs', () => {
    expect(wrapper.find('input[id="subBuildingName"]').exists()).toBeTruthy();
    expect(wrapper.find('input[id="buildingNumber"]').exists()).toBeTruthy();
    expect(wrapper.find('input[id="buildingName"]').exists()).toBeTruthy();
    expect(wrapper.find('input[id="postcode"]').exists()).toBeTruthy();
    expect(wrapper.find('input[id="streetName"]').exists()).toBeTruthy();
    expect(wrapper.find('input[id="townCity"]').exists()).toBeTruthy();
    expect(wrapper.find('input[id="postcode"]').exists()).toBeTruthy();
    expect(wrapper.find('input[id="county"]').exists()).toBeTruthy();
    expect(wrapper.find('input[id="country"]').exists()).toBeTruthy();
  });

  it('should have a for attribute on all input labels', () => {
    expect(wrapper.find('label').at(0).props()['htmlFor']).toBeDefined();
    expect(wrapper.find('label').at(0).props()['htmlFor']).toBe('subBuildingName');
    expect(wrapper.find('label').at(1).props()['htmlFor']).toBeDefined();
    expect(wrapper.find('label').at(1).props()['htmlFor']).toBe('buildingNumber');
    expect(wrapper.find('label').at(2).props()['htmlFor']).toBeDefined();
    expect(wrapper.find('label').at(2).props()['htmlFor']).toBe('buildingName');
    expect(wrapper.find('label').at(3).props()['htmlFor']).toBeDefined();
    expect(wrapper.find('label').at(3).props()['htmlFor']).toBe('streetName');
    expect(wrapper.find('label').at(4).props()['htmlFor']).toBeDefined();
    expect(wrapper.find('label').at(4).props()['htmlFor']).toBe('townCity');
    expect(wrapper.find('label').at(5).props()['htmlFor']).toBeDefined();
    expect(wrapper.find('label').at(5).props()['htmlFor']).toBe('county');
    expect(wrapper.find('label').at(6).props()['htmlFor']).toBeDefined();
    expect(wrapper.find('label').at(6).props()['htmlFor']).toBe('postcode');

  });

  it('will set the AccessibleAutoComplete component', () => {
    expect(wrapper.find('AccessibleAutoComplete').exists()).toBeTruthy();
    expect(wrapper.find('AccessibleAutoComplete[id="country"]').exists()).toBeTruthy();
  });

  it('should find the cancel and submit exporter address buttons', () => {
    expect(wrapper.find('button#cancel').exists()).toBeTruthy();
    expect(wrapper.find('button#submit').exists()).toBeTruthy();
  });

  it('should find the the fields with the preselected address', () => {
    expect(wrapper.find('input[id="subBuildingName"]').instance().value).toEqual('sub building name');
    expect(wrapper.find('input[id="buildingNumber"]').instance().value).toEqual('building number');
    expect(wrapper.find('input[id="buildingName"]').instance().value).toEqual('LANCASTER HOUSE');
    expect(wrapper.find('input[id="streetName"]').instance().value).toEqual('HAMPSHIRE COURT');
    expect(wrapper.find('input[id="townCity"]').instance().value).toEqual('NEWCASTLE UPON TYNE');
    expect(wrapper.find('input[id="county"]').instance().value).toEqual('TYNESIDE');
    expect(wrapper.find('input[id="postcode"]').instance().value).toEqual('NE4 7YH');
    expect(wrapper.find('input[id="country"]').instance().value).toEqual('ENGLAND');
  });

  it('should handle change on the fields', () => {
    wrapper.find('input[name="subBuildingName"]').simulate('change', { target: { name: 'subBuildingName', value: 'Flat 1' } });
    expect(wrapper.find('input[name="subBuildingName"]').instance().value).toEqual('Flat 1');


    wrapper.find('input[name="buildingNumber"]').simulate('change', { target: { name: 'buildingNumber', value: '11' } });
    expect(wrapper.find('input[name="buildingNumber"]').instance().value).toEqual('11');

    wrapper.find('input[name="buildingName"]').simulate('change', { target: { name: 'buildingName', value: 'Winter house' } });
    expect(wrapper.find('input[name="buildingName"]').instance().value).toEqual('Winter house');

    wrapper.find('input[name="streetName"]').simulate('change', { target: { name: 'streetName', value: 'Water of Leith' } });
    expect(wrapper.find('input[name="streetName"]').instance().value).toEqual('Water of Leith');

    wrapper.find('input[name="townCity"]').simulate('change', { target: { name: 'townCity', value: 'Edinburgh' } });
    expect(wrapper.find('input[name="townCity"]').instance().value).toEqual('Edinburgh');

    wrapper.find('input[name="county"]').simulate('change', { target: { name: 'county', value: 'Midlothian' } });
    expect(wrapper.find('input[name="county"]').instance().value).toEqual('Midlothian');

    wrapper.find('input[name="postcode"]').simulate('change', { target: { name: 'postcode', value: 'EH1 2EH' } });
    expect(wrapper.find('input[name="postcode"]').instance().value).toEqual('EH1 2EH');

    wrapper.find('input[name="country"]').simulate('change', { target: { name: 'county', value: 'Scotland' } });
    expect(wrapper.find('input[name="country"]').instance().value).toEqual('Scotland');

    wrapper.find('input[name="country"]').simulate('blur', { target: { name: 'county', value: 'Scotland' } });
    expect(wrapper.find('input[name="country"]').instance().value).toEqual('Scotland');

  });



  it('should handle blur on the fields', () => {

    wrapper.find('input[name="subBuildingName"]').simulate('blur', { target: { name: 'subBuildingName', value: 'Flat 1' } });
    expect(wrapper.find('input[name="subBuildingName"]').instance().value).toEqual('Flat 1');

    wrapper.find('input[name="buildingNumber"]').simulate('blur', { target: { name: 'buildingNumber', value: '11' } });
    expect(wrapper.find('input[name="buildingNumber"]').instance().value).toEqual('11');

    wrapper.find('input[name="buildingName"]').simulate('blur', { target: { name: 'buildingName', value: 'Winter house' } });
    expect(wrapper.find('input[name="buildingName"]').instance().value).toEqual('Winter house');

    wrapper.find('input[name="streetName"]').simulate('blur', { target: { name: 'streetName', value: 'Water of Leith' } });
    expect(wrapper.find('input[name="streetName"]').instance().value).toEqual('Water of Leith');

    wrapper.find('input[name="townCity"]').simulate('blur', { target: { name: 'townCity', value: 'Edinburgh' } });
    expect(wrapper.find('input[name="townCity"]').instance().value).toEqual('Edinburgh');

    wrapper.find('input[name="county"]').simulate('blur', { target: { name: 'county', value: 'Midlothian' } });
    expect(wrapper.find('input[name="county"]').instance().value).toEqual('Midlothian');

    wrapper.find('input[name="postcode"]').simulate('blur', { target: { name: 'postcode', value: 'EH1 2EH' } });
    expect(wrapper.find('input[name="postcode"]').instance().value).toEqual('EH1 2EH');

    wrapper.find('input[name="country"]').simulate('blur', { target: { name: 'county', value: 'Scotland' } });
    expect(wrapper.find('input[name="country"]').instance().value).toEqual('Scotland');

  });

  it('should take a snapshot of AddAddressForm: PostcodeLookUp: Step 3', ()=> {
    const { container } = render(wrapper);
    expect(container).toMatchSnapshot();
  });
});

describe('AddAddressForm without pre selected values', () => {

  const mockStore = configureStore([thunk.withExtraArgument()]);

  const store = mockStore({
  });

  const props = {
    countries: ['Brazil', 'UK'],
    errors: {},

  };

  const wrapper = mount(
    <Provider store={store}>
      <MemoryRouter>
        <AddAddressForm {...props} />
      </MemoryRouter>
    </Provider>
  );

  it('should find the the fields without the preselected address', () => {
    expect(wrapper.find('input[id="subBuildingName"]').instance().value).toEqual('');
    expect(wrapper.find('input[id="buildingNumber"]').instance().value).toEqual('');
    expect(wrapper.find('input[id="buildingName"]').instance().value).toEqual('');
    expect(wrapper.find('input[id="streetName"]').instance().value).toEqual('');
    expect(wrapper.find('input[id="townCity"]').instance().value).toEqual('');
    expect(wrapper.find('input[id="county"]').instance().value).toEqual('');
    expect(wrapper.find('input[id="postcode"]').instance().value).toEqual('');
    expect(wrapper.find('input[id="country"]').instance().value).toEqual('');
  });

  it('should render the errors when there are any', () => {
    const mockStore = configureStore([thunk.withExtraArgument()]);

    const store = mockStore({});

    const props = {
      countries: ['Brazil', 'UK'],
      errors: {
        streetNameError: 'commonWhatExportersAddressErrorStreetName',
        townCityError: 'commonWhatExportersAddressErrorTownCity',
        postcodeError: 'commonLookupAddressPageErrorPostcodeEmpty',
        countryError: 'commonWhatExportersAddressErrorCountry'
      }

    };

    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <AddAddressForm {...props} />
        </MemoryRouter>
      </Provider>
    );

    expect(wrapper.find('ErrorText').at(0).text()).toBe('Enter a street name');
    expect(wrapper.find('ErrorText').at(1).text()).toBe('Enter the town or city');
    expect(wrapper.find('ErrorText').at(2).text()).toBe('Enter a postcode');
    expect(wrapper.find('ErrorText').at(3).text()).toBe('Select a country from the list');
  });
});

describe('AddAddressForm actions', () => {

  const props = {
    countries: ['Brazil', 'UK'],
    errors: {},
    preSelected: {
      building_name: 'LANCASTER HOUSE',
      city: 'NEWCASTLE UPON TYNE',
      country: 'ENGLAND',
      county: 'TYNESIDE',
      postCode: 'NE4 7YH',
      street_name: 'HAMPSHIRE COURT',
      sub_building_name: 'sub building name',
      building_number: 'building number'
    },
    handleCancelButton: jest.fn(),
    onSubmit: jest.fn()
  };

  const mockStore = configureStore([thunk.withExtraArgument()]);

  const store = mockStore({
  });
  // const wrapper = mount(<AddAddressForm {...props}/>);

  const wrapper = mount(
    <Provider store={store}>
      <MemoryRouter>
        <AddAddressForm {...props} />
      </MemoryRouter>
    </Provider>

  );

  let mockGetElementById;


  beforeEach(() => {
    mockGetElementById = jest.spyOn(document, 'getElementById');
  });

  afterEach(() => {
    mockGetElementById.mockReset();
  });

  it('will call the on confirm of the AddressForm component', () => {
    const country = 'Nigeria';

    wrapper.setState({
      'country': 'Nigeria'
    });

    wrapper.mount();
    const accessibleAutoCompleteEl = wrapper.find('AccessibleAutoComplete[id="country"]');
    accessibleAutoCompleteEl.props().onConfirm(country);

    expect(wrapper.state().country).toBe('Nigeria');
  });

  it('will call the on confirm with no params of the AddressForm component', () => {
    const country = null;
    const el = { value: 'England' };
    wrapper.setState({
      'country': 'England'
    });

    wrapper.mount();

    mockGetElementById.mockImplementation(() => el);

    const accessibleAutoCompleteEl = wrapper.find('AccessibleAutoComplete[id="country"]');
    accessibleAutoCompleteEl.props().onConfirm(country);

    expect(wrapper.state().country).toBe('England');
  });

  it('will call the on confirm with no params and no element of the AddressForm component', () => {
    const country = null;

    wrapper.setState({
      'country': 'England'
    });

    wrapper.mount();

    mockGetElementById.mockImplementation(() => null);

    const accessibleAutoCompleteEl = wrapper.find('AccessibleAutoComplete[id="country"]');
    accessibleAutoCompleteEl.props().onConfirm(country);

    expect(wrapper.state().country).toBe('England');
  });

  it('will call the the on submit for component', () => {

    const expected =  {
      subBuildingName: 'sub building name',
      buildingNumber: 'building number',
      buildingName: 'LANCASTER HOUSE',
      streetName: 'HAMPSHIRE COURT',
      townCity: 'NEWCASTLE UPON TYNE',
      county: 'TYNESIDE',
      postcode: 'NE4 7YH',
      country: 'England',
    };

    const event = {
      preventDefault: jest.fn()
    };
    const formEl = wrapper.find('form');
    formEl.props().onSubmit(event);

    expect(formEl.exists()).toBeTruthy();
    expect(event.preventDefault).toHaveBeenCalled();
    expect(props.onSubmit).toHaveBeenCalledWith(expected);
  });
});