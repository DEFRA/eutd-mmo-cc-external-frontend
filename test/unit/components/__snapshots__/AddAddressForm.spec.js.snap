// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddAddressForm should take a snapshot of AddAddressForm: PostcodeLookUp: Step 3 1`] = `
<div>
  <form>
    <div
      class="css-1lzr9ic-StyledRow e4u1u2g0"
    >
      <div
        class="css-1w80l5h-StyledColumn e4u1u2g0"
      >
        <label
          class="e1vcssft0 e1vcssft0 css-5tg97n-StyledLabel-StyledHoc-StyledHoc e4u1u2g0"
          for="subBuildingName"
        >
          <span
            class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
          >
            Sub-building name
          </span>
          <span
            class="e1vcssft0 css-1whl0n4-StyledHint-StyledHoc e4u1u2g0"
          >
            For example, the flat or apartment number
          </span>
          
          <input
            class="mediumInput e1vcssft0 css-4n7f6s-StyledInput-StyledHoc e4u1u2g0"
            id="subBuildingName"
            name="subBuildingName"
            type="text"
            value="sub building name"
          />
        </label>
      </div>
    </div>
    <div
      class="css-1lzr9ic-StyledRow e4u1u2g0"
    >
      <div
        class="css-1w80l5h-StyledColumn e4u1u2g0"
      >
        <label
          class="e1vcssft0 e1vcssft0 css-5tg97n-StyledLabel-StyledHoc-StyledHoc e4u1u2g0"
          for="buildingNumber"
        >
          <span
            class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
          >
            Building number
          </span>
          
          <input
            class="smallInput e1vcssft0 css-4n7f6s-StyledInput-StyledHoc e4u1u2g0"
            id="buildingNumber"
            name="buildingNumber"
            type="text"
            value="building number"
          />
        </label>
      </div>
    </div>
    <div
      class="css-1lzr9ic-StyledRow e4u1u2g0"
    >
      <div
        class="css-1w80l5h-StyledColumn e4u1u2g0"
      >
        <label
          class="e1vcssft0 e1vcssft0 css-5tg97n-StyledLabel-StyledHoc-StyledHoc e4u1u2g0"
          for="buildingName"
        >
          <span
            class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
          >
            Building name
          </span>
          
          <input
            class="mediumInput e1vcssft0 css-4n7f6s-StyledInput-StyledHoc e4u1u2g0"
            id="buildingName"
            name="buildingName"
            type="text"
            value="LANCASTER HOUSE"
          />
        </label>
      </div>
    </div>
    <div
      class="css-1lzr9ic-StyledRow e4u1u2g0"
    >
      <div
        class="css-1w80l5h-StyledColumn e4u1u2g0"
      >
        <label
          class="e1vcssft0 e1vcssft0 css-5tg97n-StyledLabel-StyledHoc-StyledHoc e4u1u2g0"
          for="streetName"
        >
          <span
            class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
          >
            Street name
          </span>
          
          <input
            class="mediumInput e1vcssft0 css-4n7f6s-StyledInput-StyledHoc e4u1u2g0"
            id="streetName"
            name="streetName"
            type="text"
            value="HAMPSHIRE COURT"
          />
        </label>
      </div>
    </div>
    <div
      class="css-1lzr9ic-StyledRow e4u1u2g0"
    >
      <div
        class="css-1w80l5h-StyledColumn e4u1u2g0"
      >
        <label
          class="e1vcssft0 e1vcssft0 css-5tg97n-StyledLabel-StyledHoc-StyledHoc e4u1u2g0"
          for="townCity"
        >
          <span
            class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
          >
            Town or city
          </span>
          
          <input
            class="mediumInput e1vcssft0 css-4n7f6s-StyledInput-StyledHoc e4u1u2g0"
            id="townCity"
            name="townCity"
            type="text"
            value="NEWCASTLE UPON TYNE"
          />
        </label>
      </div>
    </div>
    <div
      class="css-1lzr9ic-StyledRow e4u1u2g0"
    >
      <div
        class="css-1w80l5h-StyledColumn e4u1u2g0"
      >
        <label
          class="e1vcssft0 e1vcssft0 css-5tg97n-StyledLabel-StyledHoc-StyledHoc e4u1u2g0"
          for="county"
        >
          <span
            class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
          >
            County/state/province
          </span>
          
          <input
            class="mediumInput e1vcssft0 css-4n7f6s-StyledInput-StyledHoc e4u1u2g0"
            id="county"
            name="county"
            type="text"
            value="TYNESIDE"
          />
        </label>
      </div>
    </div>
    <div
      class="css-1lzr9ic-StyledRow e4u1u2g0"
    >
      <div
        class="css-1w80l5h-StyledColumn e4u1u2g0"
      >
        <label
          class="e1vcssft0 e1vcssft0 css-5tg97n-StyledLabel-StyledHoc-StyledHoc e4u1u2g0"
          for="postcode"
        >
          <span
            class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
          >
            Postcode
          </span>
          
          <input
            class="smallInput e1vcssft0 css-4n7f6s-StyledInput-StyledHoc e4u1u2g0"
            id="postcode"
            name="postcode"
            type="text"
            value="NE4 7YH"
          />
        </label>
      </div>
    </div>
    <div
      class="css-1lzr9ic-StyledRow e4u1u2g0"
    >
      <div
        class="css-1w80l5h-StyledColumn e4u1u2g0"
      >
        <label
          class="e1vcssft0 css-137pbp-StyledLabel-StyledHoc e4u1u2g0"
          for="Country"
        >
          Country
          <div
            style="max-width: 23em;"
          >
            <span
              class="e1vcssft0 css-9vo93-StyledError-StyledHoc e4u1u2g0"
            />
            <div
              class="autocomplete__wrapper"
            >
              <div
                style="border: 0px; height: 1px; margin-bottom: -1px; margin-right: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
              >
                <div
                  aria-atomic="true"
                  aria-live="polite"
                  id="country__status--A"
                  role="status"
                />
                <div
                  aria-atomic="true"
                  aria-live="polite"
                  id="country__status--B"
                  role="status"
                />
              </div>
              
              <input
                aria-activedescendant="false"
                aria-autocomplete="list"
                aria-describedby="country__assistiveHint"
                aria-expanded="false"
                aria-owns="country__listbox"
                autocomplete="off"
                class="autocomplete__input autocomplete__input--default"
                id="country"
                name="country"
                placeholder=""
                role="combobox"
                type="text"
                value="ENGLAND"
              />
              <ul
                class="autocomplete__menu autocomplete__menu--overlay autocomplete__menu--hidden"
                id="country__listbox"
                role="listbox"
              >
                <li
                  aria-posinset="1"
                  aria-selected="false"
                  aria-setsize="1"
                  class="autocomplete__option"
                  id="country__option--0"
                  role="option"
                  tabindex="-1"
                >
                  ENGLAND
                </li>
              </ul>
              <span
                id="country__assistiveHint"
                style="display: none;"
              >
                When autocomplete results are available use up and down arrows to review and enter to select.  Touch device users, explore by touch or with swipe gestures.
              </span>
            </div>
          </div>
        </label>
      </div>
    </div>
    <div
      class="css-1lzr9ic-StyledRow e4u1u2g0"
    >
      <button
        class="css-17uw1q2-StyledButton-StyledHoc e4u1u2g0"
        id="cancel"
        name="cancel"
        type="button"
        value="cancel"
      >
        Cancel
      </button>
      <div
        class="css-1w80l5h-StyledColumn e4u1u2g0"
      >
        <button
          class="button"
          id="submit"
        >
          Continue
        </button>
      </div>
    </div>
  </form>
</div>
`;
